pennylane/transforms/unbroadcast_expand.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the tape expansion function for unbroadcasting a
+broadcasted tape into multiple tapes."""
+import pennylane as qml
+from pennylane.transforms.batch_transform import batch_transform
+
+
+@batch_transform
+def unbroadcast_expand(tape):
```

Comment:
minor comment, but I must admit I find the name hard to parse! Would the following do a better job of describing what the transform does?

- `broadcast`
- `broadcast_parameters`
- `parameter_expand`

In my mind, broadcasting is almost the _act_ of taking one tape with ndim parameters and creating multiple tapes 🤔 So 'unbroadcasting' is hard for me to parse

Reply:
> In my mind, broadcasting is almost the act of taking one tape with ndim parameters and creating multiple tapes 
Ah I see! I'm not sure, which name fits best here. In docs and comments I have been using "broadcasted tape" for tapes with `batch_size!=None`. Maybe `broadcast_expand` actually is a good name? It makes clear that it is in the context of broadcasting and expands into multiple tapes. What do you think?